snap/validate.go

```diff
@@ -35,3 +35,47 @@ func ValidateName(name string) error {
 	}
 	return nil
 }
+
+// Validate verifies the content in the info.
+func Validate(info *Info) error {
+	return nil
+}
+
+func validateField(name, cont string, whitelist *regexp.Regexp) error {
+	if !whitelist.MatchString(cont) {
+		return fmt.Errorf("app description field '%s' contains illegal %q (legal: '%s')", name, cont, whitelist)
+
+	}
+	return nil
+}
+
+// appContentWhitelist is the whitelist of legal chars in the "apps"
+// section of snap.yaml
+var appContentWhitelist = regexp.MustCompile(`^[A-Za-z0-9/. _#:-]*$`)
+
+// ValidateApp verifies the content in the app info.
+func ValidateApp(app *AppInfo) error {
```

Comment:
Why don't we just have ``func (app *AppInfo) Validate() error`` instead? Then we could simply recursively validate everything from snap.Info